### PR TITLE
MM-47250: Add a new section in system console for products

### PR DESCRIPTION
**Pull request by agnivade**
For the MPA, we add a new section which will contain
product specific settings.

https://mattermost.atlassian.net/browse/MM-47250

### CODE REVIEW

**Comment by agnivade** [comment]
@hmhealey - One thing I need help with is - if I start typing "boards" in the system console text box, then the boards section doesn't show up. What do I need to do for it to happen?

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -5769,6 +5769,43 @@ const AdminDefinition = {
...
+            <CogOutlineIcon
+                size={16}
+                className={'category-icon fa'}
+                color={'currentColor'}
+            />
+        ),
+        sectionTitle: t('admin.sidebar.products'),
+        sectionTitleDefault: 'Products',
+        isHidden: it.any(
+            it.configIsFalse('FeatureFlags', 'BoardsProduct'),
```
Comment (agnivade) [inline components/admin_console/admin_definition.jsx:5783]: @hmhealey - It seems like our test case suite somehow doesn't like `it.configIsFalse` under the main products section. But the code works, and the section indeed gets hidden. If I move the function under `settings [`, the tests pass.

I can't figure out what needs to change in the tests for this to pass. Any pointers would be very appreciated.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Comment by agnivade** [comment]
@hmhealey - If you are testing something,  you'd also need to set the feature flag `BoardsProduct` as as true for it to appear. Not sure if it's possible via the GitHub tag UI. You'll need to use the `/cloud` command for that.

**Comment by hmhealey** [comment]
I'm trying to test this locally, and I'm not having the new section show up even with the environment variable set.

I'm also seeing some errors about Boards failing to load though ([screenshot](https://user-images.githubusercontent.com/3277310/195929149-1befb8d4-957d-4872-813f-a522601b35d1.png)), but I don't think that should be related to this code not working

**Comment by agnivade** [comment]
Are you sure you are running this branch (https://github.com/mattermost/mattermost-server/pull/21333) along with your webapp branch? Without that the permissions won't be applied and you won't see it.

**Review by hmhealey — APPROVED** [review]
Turns out that those tests were failing because we had some bad typing in a test, so that caused errors at runtime. And for the new settings not showing up in search, the new product section needed to be added to the search index

**Pull request merged**